grub-core/loader/multiboot_mbi2.c

```diff
@@ -714,15 +764,20 @@ grub_multiboot2_make_mbi (grub_uint32_t *target)
   COMPILE_TIME_ASSERT (MULTIBOOT_TAG_ALIGN % sizeof (grub_properly_aligned_t) == 0);
 
   err = grub_relocator_alloc_chunk_align (grub_multiboot2_relocator, &ch,
-					  MBI_MIN_ADDR, UP_TO_TOP32 (bufsize),
+					  MBI_MIN_ADDR, 0xffffffff - bufsize,
```

Comment:
Let's not change it. UP_TO_TOP32(value) is probably doing the same as `0xffffffff - value`

Reply:
@miczyg1 macro used https://github.com/TrenchBoot/grub/pull/8/commits/c3f98c34e3111453bfdd7a2f57adb03bcd848453#diff-285279252527f5bc911013189ac09a0880271f60dc8f28f28fe68c2f5d517c74R767